networking_generic_switch/devices/netmiko_devices/__init__.py

```diff
@@ -136,7 +176,14 @@ def send_commands_to_device(self, cmd_set):
             with ngs_lock.PoolLock(self.locker, **self.lock_kwargs):
                 with self._get_connection() as net_connect:
                     net_connect.enable()
-                    output = net_connect.send_config_set(
+
+                    # output = net_connect.send_config_set(
+                    #     config_mode_cmd='configure private',
+                    #     config_commands=cmd_set)
+                    # FIXME: use the above, commented out version of
+                    # send_config_set once netmiko patch is accepted upstream.
```

Comment:
I would try just adding this before the existing send_config_set(), I think that should be OK, because it only activates config mode if its not already enabled:

`net_connect.config_mode(config_command='configure private')`

Reply:
technically we would need to copy the save_configuration pattern and add something like start_config_transaction() that is a no op for all drivers except the juniper.

I would have much more hope of getting the above accepted upstream, if that does turn out to work.

Reply:
If we want to go this route, we would need a new driver method I think, the default could call net_connect.send_config_set and the juniper driver can override that method with the above changes.

Having said that, I still think my preference is trying an explicit start configuration call that only gets made by the juniper.

Reply:
Just adding net_connect.config_mode(config_command='configure private') would be the ideal solution, I agree, but that's the problem.
As it stands, the send_config_set method in netmiko has a line that reads: output = self.config_mode()
so this will override any value provided (such as 'configure private') and revert to the default for the driver. In the case of juniper, that's ''.
This is why I have had to hack the whole send_config_set method, above.

It's a good point about making this juniper specific though. That was an oversight, I'll fix that now.